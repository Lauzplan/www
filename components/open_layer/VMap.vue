<template>
  <div style="height: 100%; width: 100%">
    <slot v-if="map"></slot>
  </div>
</template>
<script>
import Map from 'ol/Map'
import View from 'ol/View'
import { getArea } from 'ol/extent'
import proj4 from 'proj4'
import { register } from 'ol/proj/proj4'
import Projection from 'ol/proj/Projection'

export default {
  name: 'VMap',
  props: {
    fit: {
      type: Array,
      default: null,
    },
  },
  provide() {
    return {
      getMapInstance: () => this.map,
    }
  },
  data() {
    return {
      map: null,
      resizeObserver: new ResizeObserver(() => this.map.updateSize()),
    }
  },
  watch: {
    fit: 'fitHandler',
  },
  mounted() {
    proj4.defs(
      'EPSG:31370',
      '+proj=lcc +lat_1=51.16666723333333 +lat_2=49.8333339 +lat_0=90 +lon_0=4.367486666666666 +x_0=150000.013 +y_0=5400088.438 +ellps=intl +towgs84=-106.869,52.2978,-103.724,0.3366,-0.457,1.8422,-1.2747 +units=m +no_defs'
    )
    register(proj4)
    this.map = new Map({
      target: this.$el,
      view: new View({
        center: [0, 0],
        zoom: 0,
        projection: new Projection({
          code: 'EPSG:31370',
          extent: [
            42000.1053417176,
            19999.916642703116,
            296000.1053417176,
            167999.91664270312,
          ],
          units: 'm',
        }),
      }),
    })
    this.resizeObserver.observe(this.$el)
  },
  beforeDestroy() {
    this.resizeObserver.disconnect()
  },
  methods: {
    fitHandler(val) {
      if (!val || getArea(val) === 0) return

      this.map.getView().fit(val, {
        duration: 500,
        padding: [20, 20, 20, 20],
      })
    },
  },
}
</script>
