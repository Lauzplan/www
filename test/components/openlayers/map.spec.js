import { render } from '@testing-library/vue'
import VMap from '@/components/open_layer/VMap.vue'

describe('VMap', () => {
  test('should render the map viewport', () => {
    const { container } = render(VMap)
    expect(container).toMatchSnapshot()
  })

  test.skip('should provide a map instance', () => {
    const { container } = render(VMap, {
      slots: {
        default: {
          render() {
            return null
          },
          inject: ['getMapInstance'],
        },
      },
    })

    expect(container).toMatchSnapshot()
  })
})
