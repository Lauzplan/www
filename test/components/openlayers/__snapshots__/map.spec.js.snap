// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VMap should render the map viewport 1`] = `
<div>
  <div
    style="height: 500px;"
  >
    <!---->
    <div
      class="ol-viewport"
      style="position: relative; overflow: hidden; width: 100%; height: 100%;"
    >
      <div
        class="ol-unselectable ol-layers"
        style="position: absolute; width: 100%; height: 100%; z-index: 0;"
      />
      <div
        class="ol-overlaycontainer"
        style="position: absolute; z-index: 0; width: 100%; height: 100%; pointer-events: none;"
      />
      <div
        class="ol-overlaycontainer-stopevent"
        style="position: absolute; z-index: 0; width: 100%; height: 100%; pointer-events: none;"
      >
        <div
          class="ol-zoom ol-unselectable ol-control"
          style="pointer-events: auto;"
        >
          <button
            class="ol-zoom-in"
            title="Zoom in"
            type="button"
          >
            +
          </button>
          <button
            class="ol-zoom-out"
            title="Zoom out"
            type="button"
          >
            −
          </button>
        </div>
        <div
          class="ol-rotate ol-unselectable ol-control ol-hidden"
          style="pointer-events: auto;"
        >
          <button
            class="ol-rotate-reset"
            title="Reset rotation"
            type="button"
          >
            <span
              class="ol-compass"
            >
              ⇧
            </span>
          </button>
        </div>
        <div
          class="ol-attribution ol-unselectable ol-control ol-collapsed"
          style="pointer-events: auto;"
        >
          <ul />
          <button
            title="Attributions"
            type="button"
          >
            <span>
              i
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
